proxy/healthy_endpoints_test.go

```diff
@@ -125,26 +137,38 @@ func TestPHCForMultipleHealthyEndpoints(t *testing.T) {
 	}
 
 	for _, algorithm := range []string{"random", "consistentHash", "roundRobin", "powerOfRandomNChoices"} {
-		t.Run(algorithm, func(t *testing.T) {
-			endpointRegistry := defaultEndpointRegistry()
-			doc := fmt.Sprintf(`* -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s">`,
-				algorithm, services[0].URL, services[1].URL, services[2].URL)
-
-			tp, err := newTestProxyWithParams(doc, Params{
-				EnablePassiveHealthCheck: true,
-				EndpointRegistry:         endpointRegistry,
-			})
-			if err != nil {
-				t.Fatal(err)
-			}
-			defer tp.close()
-
-			ps := httptest.NewServer(tp.proxy)
-			defer ps.Close()
+		balanceFactors := []bool{false}
+		if algorithm == "consistentHash" {
+			balanceFactors = []bool{false, true}
+		}
 
-			failedReqs := sendGetRequests(t, ps)
-			assert.Equal(t, 0, failedReqs)
-		})
+		for _, balanceFactor := range balanceFactors {
+			t.Run(fmt.Sprintf("%s_%t", algorithm, balanceFactor), func(t *testing.T) {
+				endpointRegistry := defaultEndpointRegistry()
+
+				balanceFactorStr := ""
+				if balanceFactor {
+					balanceFactorStr = ` -> consistentHashBalanceFactor(1.25)`
+				}
+				doc := fmt.Sprintf(`* -> consistentHashKey("${request.header.ConsistentHashKey}") %s -> <%s, "%s", "%s", "%s">`,
+					balanceFactorStr, algorithm, services[0].URL, services[1].URL, services[2].URL)
+
+				tp, err := newTestProxyWithParams(doc, Params{
+					EnablePassiveHealthCheck: true,
+					EndpointRegistry:         endpointRegistry,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				defer tp.close()
+
+				ps := httptest.NewServer(tp.proxy)
+				defer ps.Close()
```

Comment:
This could be a helper that sets up backends and receives a route template to make more readable.
Then each combination could be a nested t.Run:
```go
t.Run("consistent hash with balance factor", func(t *testing.T) {
  setupProxy(t, `* -> consistentHashKey("${request.header.ConsistentHashKey}") -> consistentHashBalanceFactor(1.25) -> <consitentHash, "%s", "%s", "%s">`, services[0].URL, services[1].URL, services[2].URL)
  failedReqs := sendGetRequests(t, ps)
  assert.Equal(t, 0, failedReqs)
})
```

Reply:
It seems to be slightly more compact, but I'm not sure which code looks better. We can continue discussion here not to block deploying fix:
https://github.com/zalando/skipper/pull/2999